index.js

```diff
@@ -1,44 +1,26 @@
-import React, { Component } from 'react';
-import { AppRegistry, Button, SectionList, StyleSheet, Text, View } from 'react-native';
+// @ts-check
```

Comment:
I'd pull all this code into a file in ./src and just use leave index.js as an entry point into that new file. That way all your code lives in ./src.